src/wallet/wallet.cpp

```diff
@@ -35,6 +35,8 @@
 #include <boost/algorithm/string/replace.hpp>
 #include <boost/thread.hpp>
 
+/** Protects the vpwallets vector */
+CCriticalSection cs_wallets;
 std::vector<CWalletRef> vpwallets;
```

Comment:
I guess this belongs to `walletinit.h` now?

Reply:
Tend to agree with @jonasschnelli here, though I agree the barrier isn't exactly clear (yet). Ideally we'd move more towards clarifying the interface here - making CWallet less and less aware of all the things around it (including other wallets).